tests/functional/object/mcg/test_bucketclass.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.bucket_utils import write_random_test_objects_to_bucket
+from ocs_ci.ocs.resources.pod import get_pod_logs, get_noobaa_operator_pod
+
+logger = logging.getLogger(__name__)
+
+
+class TestBucketClass:
+
+    def test_bucketclass_modification(
+        self,
+        bucket_factory,
+        backingstore_factory,
+        awscli_pod,
+        test_directory_setup,
+        mcg_obj,
+    ):
+        """
+        Test to verify bucket class placement policy modification
+        from 'Spread' to 'Mirror'
+
+        """
+
+        # create a bucket
+        bucketclass_dict = {
+            "interface": "CLI",
+            "backingstore_dict": {"aws": [(1, "eu-central-1")]},
+        }
+        bucket = bucket_factory(interface="CLI", bucketclass=bucketclass_dict)[0]
+
+        # write some object data to the bucket
+        logger.info("Writing some objects to the bucket")
+        write_random_test_objects_to_bucket(
+            awscli_pod,
+            bucket.name,
+            test_directory_setup.origin_dir,
+            amount=1,
+            mcg_obj=mcg_obj,
+        )
+
+        # create new backingstore
+        logger.info("Creating a new backingstore")
+        backingstore = backingstore_factory(
+            method="CLI",
+            uls_dict=bucketclass_dict["backingstore_dict"],
+        )[0]
+
+        # modify the existing bucketclass placement policy to Mirror from Spread
+        # and add new backingstore under backingstores
+        logger.info("Updating bucketclass")
+        obc_obj = OCP(
+            kind=constants.BUCKETCLASS,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=bucket.bucketclass.name,
+        )
+        obc_obj.patch(
+            params=f'[{{"op": "add", "path": "/spec/placementPolicy/tiers/0/backingStores/-", '
+            f'"value": "{backingstore.name}"}},'
+            f' {{"op": "replace", "path": "/spec/placementPolicy/tiers/0/placement", "value": "Mirror"}}]',
+            format_type="json",
+        )
+
```

Comment:
It might not be in the reproduction steps in the original BZ, but please also check that the original bucket is still usable after updating the Bucketclass.

Repeating `write_random_test_objects_to_bucket` on it should be enough.